### PR TITLE
Dpi aware

### PR DESCRIPTION
**Pull request by jbialobr**
@KindDragon 
Changes moved from master.
The proposed solution doesn't work. Encountered problems:
1. FormBrowse: tabs become invisible, revisions take the whole client area.
2. FormCheckout branch is cut off.
gitext

### CODE REVIEW

**Comment by KindDragon** [comment]
Hmm... Can you move only 472d61b ?

**Comment by KindDragon** [comment]
Works for me without your issues

**Comment by jbialobr** [comment]
After removing 472d61b it looks the same:
gitext

**Comment by KindDragon** [comment]
> FormBrowse: tabs become invisible, revisions take the whole client area.

https://github.com/gitextensions/gitextensions/pull/3651/commits/05090a0b6550a0edd23e6f31af9d6a1cf7b8d99d#diff-6d931ecb996aeb8676180da98e3f6861R2801
I think somehow spliter distance changed to incorrect value for you. It should not change at the first start.

> FormCheckout branch is cut off.

I rewrote commit "Save form current DPI with form position/size and with SplitterDistance", should probably fix this.

**Comment by jbialobr** [comment]
After your last commits FormCheckout  looks good and FormCommit works as expected, but FormBrowse changes the splitter position like FormCommit did before your change:)

**Comment by jbialobr** [comment]
I did the same for FormBrows as in cf28d541c9a010e639cfeee5068bfb2258fe16f0 and now it works for me.

File: GitUI/CommandsDialogs/FormBrowse.cs [hunk 1]
```diff
@@ -2783,6 +2782,12 @@ private void DiffContextMenu_Opening(object sender, System.ComponentModel.Cancel
             }
         }
 
+        protected override void RestorePosition(string name)
+        {
+            base.RestorePosition(name);
```
Comment (KindDragon) [inline GitUI/CommandsDialogs/FormBrowse.cs]: Don't you mind if I change it to work the same way as in FormCommit? FormCommit use event OnLoad to restore splitter position after restoring form size

**Comment by KindDragon** [comment]
Thank you for fixes!

**Comment by KindDragon** [comment]
What was issue with `Dashboard` splitter?

**Comment by jbialobr** [comment]
It was restoring splitter positions on each refresh call, so when you went to dashboard, changed a splitter, went to a repo and again went to the dashboard then your changes in splitters were overwriten.

**Comment by mscrivo** [comment]
Have you guys considered adding the new per monitor DPI awareness flags?  

```
<asmv3:application>
  <asmv3:windowsSettings xmlns="http://schemas.microsoft.com/SMI/2005/WindowsSettings">
    <dpiAware>True/PM</dpiAware>
    <dpiAwareness xmlns="http://schemas.microsoft.com/SMI/2016/WindowsSettings">PerMonitorV2,PerMonitor</dpiAwareness>
  </asmv3:windowsSettings>
</asmv3:application>
```

File: GitUI/CommandsDialogs/FormBrowse.cs (hunk 1 above)
Comment (jbialobr) [inline GitUI/CommandsDialogs/FormBrowse.cs]: It is fine by me.

**Comment by KindDragon** [comment]
@mscrivo I tried this, this doesn't work well with our forms.

**Comment by mscrivo** [comment]
ah that's unfortunate .. what issues did you see out of curiosity?  I have a mix of monitors with different scaling factors, I might try it out and see how it works.

**Comment by mscrivo** [comment]
@KindDragon I just tried with the above settings, and the only forms that I can see that are messed up are some of the settings dialogs and the pull dialog.  But those are messed up on my 4K monitor @150% scaling whether per monitor DPI awareness is on or off.   However, with the per monitor flags, it scales properly when moving between my two monitors which are different resolution and different scaling factors.  It seems some work needs to be done on those dialogs regardless to fix the autoscaling.

**Comment by KindDragon** [comment]
> However, with the per monitor flags, it scales properly when moving between my two monitors which are different resolution and different scaling factors. It seems some work needs to be done on those dialogs regardless to fix the autoscaling.

It's provide ugly artefacts after moving window and sometimes even GitExt crash for me.

**Pull request merged**